README.md

```diff
@@ -6,26 +6,81 @@ Implements https://groups.google.com/forum/#!msg/kubernetes-sig-api-machinery/rH
 
 ## Purpose
 
-This controller implements the `EtcdStorage` type, used to provide etcd storage for aggregated API servers.
+This controller implements the `EtcdStorage` type, used to provide etcd storage for the aggregated API servers.
 
 ## Compatibility
 
 HEAD of this repo matches versions `1.10` of `k8s.io/apiserver`, `k8s.io/apimachinery`, and `k8s.io/client-go`.
 
-## Running
+## Running controller
 
-Prerequisite: Since the sample-controller uses apps/v1 deployments, the Kubernetes cluster version should be greater than 1.9.
+There're two ways to run `EtcdProxyController`:
 
+* in-cluster, which is done by deploying [the deployment manifest, located in the `artifacts/deployment` directory](https://github.com/xmudrii/etcdproxy-controller/blob/master/artifacts/deployment/00-etcdproxy-controller.yaml),
+* out-of-cluster, which is done by building the controller binary and running it on the local machine.
+
+Before running the controller, you need to deploy the etcd cluster. There's an example manifests along with the deploying instructions in the [`artifacts/etcd` directory](https://github.com/xmudrii/etcdproxy-controller/tree/master/artifacts/etcd).
+
+The deployment manifest assumes etcd is available on `https://etcd-svc-1.etcd.svc:2379`. This can be configured by updating the [`00-etcdproxy-controller.yaml` manifest](https://github.com/xmudrii/etcdproxy-controller/blob/master/artifacts/deployment/00-etcdproxy-controller.yaml).
+
+When running out-of-cluster, the etcd URL can be changed using the `etcd-core-url` flag.
+
+### Providing the core etcd client certificates to the controller
+
+When deploying the core etcd using the provided manifests, you can deploy the client certificates using the `etcd-client-certs.yaml` manifest. [The README of `artifacts/etcd`](https://github.com/xmudrii/etcdproxy-controller/tree/master/artifacts/etcd) contains more details about deploying the client certificates.
+
+If you're deploying etcd using another method, the `EtcdProxyController` requires you to provide it the CA certificate (`ca.pem`) as a ConfigMap, and the client certificate (`client.pem`) and key (`client-key.pem`) as
+a Secret.
+
+Currently, both the ConfigMap and Secret must be located in the controller namespace.
+
+The ConfigMap is by default called `etcd-coreserving-ca`, and the name can be changed using the `--etcd-core-ca-configmap` flag.
```

Comment:
why is called `... coreserving ... `? Sounds confusing. `etcd-core-ca` and `etcd-core-client-cert` would make more sense, no?

Reply:
Makes sense. As I'm trying to focus this PR on docs-only changes, let's leave it for another PR if that's okay. We can also discuss this on-call today.